tests/integration/test_core.py

```diff
@@ -366,114 +473,43 @@ def patched_getaddrinfo(*args):
+    assert: A Kubernetes ingress modsecurity should be enabled and proper rules should be set up
+        for WordPress.
     """
     assert ops_test.model
-    swift_conn = swiftclient.Connection(
-        authurl=openstack_environment["OS_AUTH_URL"],
-        auth_version="3",
-        user=openstack_environment["OS_USERNAME"],
-        key=openstack_environment["OS_PASSWORD"],
-        os_options={
-            "user_domain_name": openstack_environment["OS_USER_DOMAIN_ID"],
-            "project_domain_name": openstack_environment["OS_PROJECT_DOMAIN_ID"],
-            "project_name": openstack_environment["OS_PROJECT_NAME"],
-        },
-    )
-    container_exists = True
-    container = "WordPress"
-    try:
-        swift_conn.head_container(container)
-    except swiftclient.exceptions.ClientException as exc:
-        if exc.http_status == 404:
-            container_exists = False
-        else:
-            raise exc
-    if container_exists:
-        for swift_object in swift_conn.get_container(container, full_listing=True)[1]:
-            swift_conn.delete_object(container, swift_object["name"])
-        swift_conn.delete_container(container)
-    swift_conn.put_container(container)
-    swift_service = swiftclient.service.SwiftService(
-        options=dict(
-            auth_version="3",
-            os_auth_url=openstack_environment["OS_AUTH_URL"],
-            os_username=openstack_environment["OS_USERNAME"],
-            os_password=openstack_environment["OS_PASSWORD"],
-            os_project_name=openstack_environment["OS_PROJECT_NAME"],
-            os_project_domain_name=openstack_environment["OS_PROJECT_DOMAIN_ID"],
-        )
-    )
-    swift_service.post(container=container, options={"read_acl": ".r:*,.rlistings"})
     application = ops_test.model.applications[application_name]
-    await application.set_config(
-        {
-            "wp_plugin_openstack-objectstorage_config": json.dumps(
-                {
-                    "auth-url": openstack_environment["OS_AUTH_URL"] + "/v3",
-                    "bucket": container,
-                    "password": openstack_environment["OS_PASSWORD"],
-                    "object-prefix": "wp-content/uploads/",
-                    "region": openstack_environment["OS_REGION_NAME"],
-                    "tenant": openstack_environment["OS_PROJECT_NAME"],
-                    "domain": openstack_environment["OS_PROJECT_DOMAIN_ID"],
-                    "swift-url": swift_conn.url,
-                    "username": openstack_environment["OS_USERNAME"],
-                    "copy-to-swift": "1",
-                    "serve-from-swift": "1",
-                    "remove-local-file": "0",
-                }
-            )
-        }
-    )
-    await ops_test.model.wait_for_idle()
+    await application.set_config({"use_nginx_ingress_modsec": "true"})
+    await ops_test.model.wait_for_idle(status=ops.model.ActiveStatus.name)  # type: ignore
```

Comment:
Out of curiosity, what is the typing issue here? is it `ops.model.ActiveStatus.name`?

Reply:
Yes, `error: Cannot determine type of "name"  [has-type]`

The `name` class attribute is defined as `NotImplemented` in base class and overwritten to string in subclass `ops.model.ActiveStatus`. mypy has some problems handling this case.